src/item.cpp

```diff
@@ -197,25 +197,15 @@ item::item(JsonObject &jo)
     deserialize(jo);
 }
 
-item::~item()
-{
-}
-
-void item::make( const std::string new_type, bool scrub )
+item& item::convert( const itype_id& new_type )
 {
     type = find_type( new_type );
-    contents.clear();
-
-    if (scrub) {
-        components.clear();
-        charges = -1;
-        bday = 0;
-        name = "";
-        curammo = NULL;
+    if( is_null() ) {
```

Comment:
Have you tested this branch? `find_type` calls `item_factory::find_template`, which always returns an item type with the matching id. If there is no such type defined, it creates one and sets its id accordingly.

You should only reach this if `new_type` is "null", which you can check upfront.

In fact, I would consider this message (and branch) here redundant: if the item type id is invalid, `item_factory::find_template` will complain on its own, no need to replicate this here.